pilot/pkg/xds/ads.go

```diff
@@ -789,14 +786,13 @@ func (s *DiscoveryServer) startPush(req *model.PushRequest) {
 
 	// Push config changes, iterating over connected envoys. This cover ADS and EDS(0.7), both share
 	// the same connection table
-	s.adsClientsMutex.RLock()
 
 	// Create a temp map to avoid locking the add/remove
-	pending := make([]*Connection, 0, len(s.adsClients))
-	for _, v := range s.adsClients {
+	clients := s.Clients()
+	pending := make([]*Connection, 0, len(clients))
```

Comment:
think this should be `s.adsClientsCount()` without the surrounding lock

Reply:
ah ignore that, misread the line 👍

Reply:
No I think you are right? We are copying it here, but Clients already copies